### ISSUE TITLE
Modrinth app downloads alpha versions automaticly

### ISSUE THREAD

**Issue by ubionexd** [modrinth/code#618]
### Is your suggested enhancement related to a problem? Please describe.

The Modrinth App should not download alpha versions of a modpack automaticly. For example: I started the app, and in the featured packs section, installed Simply Optimized. When I launched it, it errored out. Turns out, the app installed the alpha version of Simply Optimized automaticly, which is for the 23w32a snapshot, and there was no indication it would download that version, only the small "Fabic 23w32a" text under the instance image, which could be confusing to casual players.

### Describe the solution you'd like

Only install the latest stable release of the modpack when clicking on the download button and make the alpha/beta versions available in the versions menu, or make a version selection pop-up when clicking install, just like when you can choose between Stable, Latest, or Other mod loader version when creating a custom pack.

### Describe alternatives you've considered

Make it more clear that you are downloading an Alpha or Beta version of the modpack.

**Comment by HyperSoop** [comment]
I second this ^

**Comment by Niels-NTG** [comment]
I've experienced a similar issue when creating a new instance for a single mod, it always installs the version of the mod that was released most recently. In this case it was a recently released patch for Minecraft/Forge 1.19.2, while the there is also a more modern version for Minecraft/Forge 1.20.2 which was also marked as the "featured release". I understand that going on version number tags alone may not work for Modrinth, since not every mod author adherses to the SemVer or CalVer conventions, but I would at least expect Theseus to install the featured version and/or the version on the release channel for the most recent Minecraft version.

**Comment by nx5314** [comment]
It does seem like a critical oversight to have implemented Release/Beta/Alpha channels for versions, but then have them be ultimately useless when updating items from the content list.  A user can install a Release version, then as soon as they go to their instance and "update all" they've been bumped to an alpha version they might have intentionally avoided.

**Comment by FeldrinH** [comment]
> It does seem like a critical oversight to have implemented Release/Beta/Alpha channels for versions, but then have them be ultimately useless when updating items from the content list.

I agree in principle, but in practice the trouble is that release channels are used in wildly different ways by different mod authors. The meaning of alpha versions between different mods can be anywhere from "stable, but a little less tested than a release version" to "half the features are being rewritten and this version should only be used if you want to help with early testing".

**Comment by FeldrinH** [comment]
Though this might be a problem that fixes itself over time if automatic installation and updating has a specific behavior for alpha versions. E.g. if Modrinth only offers alpha versions if players explicitly opt in then authors will probably start publishing alpha versions only when they want players to explicitly opt-in.

**Comment by Niels-NTG** [comment]
> > It does seem like a critical oversight to have implemented Release/Beta/Alpha channels for versions, but then have them be ultimately useless when updating items from the content list.
> 
> I agree in principle, but in practice the trouble is that release channels are used in wildly different ways by different mod authors. The meaning of alpha versions between different mods can be anywhere from "stable, but a little less tested than a release version" to "half the features are being rewritten and this version should only be used if you want to help with early testing".

True, but I don't think it's a problem if Modrinth would be opinionated about how to label versions. Which is not to say that mod authors should be forced to use SemVer or CalVer. There should be some way to mark a release as latest stable and/or latest unstable, instead of relying on whichever version was uploaded most recently.

**Comment by Prospector** [comment]
I agree that it would have been wise to decide on a standard for how each release channel should behave from the beginning, but the truth is we're in too deep to enforce much there without causing issues for someone else.

Will continue to think on it, and perhaps we could add new release channels that actually do have semantic meaning? Not sure

**Comment by Prospector** [comment]
Perhaps a project-wide flag you could set for whether or not your projects should use semantic release channels?

**Comment by nx5314** [comment]
I think there are strategies that could be implemented to allow for both the user and author to determine updating protocols to their preferences since we already have the channels in place.  On the user-side there might be an instance-wide setting for whether updates of a certain channel are allowed, or if more granular control is viable then per-mod settings.  If an author oriented approach is more in line with the team's vision then I would agree a simple flag for whether different channels are pushed as automatic updates or not would be pretty straight forward.

Although I might argue user-side settings could still be useful if some mod authors were to continue to handle their channels differently than a user would prefer.  Either way would be a much appreciated improvement on the system.

I do think there's merit to adding release channel management options given that, at least in my experience, authors that do utilise release channels tend to do so with the intent that users will be choosing which branch to select and often add notes or warnings in their changelogs, so at least some authors would likely be happy to flag a build or channel as "don't auto update".

**Comment by FeldrinH** [comment]
Recently encountered an issue where this would have helped. The latest Sodium version for 1.21.1 is marked as alpha and it turns out for good reason: it has quite a lot of compatibility issues. Opting out of alpha/unstable releases would have saved me quite a bit of headache tracking this down.

**Comment by FeldrinH** [comment]
Also, over time I've noticed that most mods do use the channels quite consistently: release tends to be stable, beta tends to be stable but less tested, and alpha tends to be unstable. I think excluding alpha versions from the list by default would be a good balance.

**Comment by Prospector** [comment]
no longer
